File: crates/flowrt-codegen/tests/golden/iox2_service_rust/expected/app/stubs/rust/planner.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Planner;

impl flowrt_app::components::Planner for Planner {
    fn on_tick(
        &mut self,
        plan: &flowrt_app::components::ServiceClient_planner_plan,
    ) -> flowrt::Status {
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Planner::default()))
}

File: crates/flowrt-codegen/tests/golden/bounded_service_iox2_cpp/expected/app/stubs/cpp/planner.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Planner final : public flowrt_app::PlannerInterface {
public:
    flowrt::Status on_tick(
        flowrt_app::ServiceClient_planner_plan& plan) override {
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Planner>());
}

}  // namespace flowrt_user
